Bug fix: Dashboard Overview, Stat Joining

c0ffeec0der opened the pull request:
Problem :
does not show new pod joining

How to simulate:
1. Deploy using scylla operator for 1 pod
2. Add 1 pod

Root cause:
Arithmetic addition in prometheus expr is not working if one condition is empty. Only work when both condition return value.

Fix:

Original code
>count(scylla_node_operation_mode==1) +  count(scylla_node_operation_mode==2)OR vector(0)

Fixing code
>count(scylla_node_operation_mode<=2)OR vector(0)

Question:
1. Is this fix is acceptable? If yes, I will update the other related dashboard templates
2. Any script to run after updating the templates? I ran 'generate-dashboard.sh', it updated the files in grafana/build

Commit:
Joining stat

Problem :
does not show new pod joining

How to simulate:
1. Deploy using scylla operator for 1 pod
2. Add 1 pod

Root cause:
Arithmetic addition in prometheus expr is not working if one condition is empty. Only work when both condition return value.

Fix:

Original code
>count(scylla_node_operation_mode==1) +  count(scylla_node_operation_mode==2)OR vector(0)

Fixing code
>count(scylla_node_operation_mode<=2)OR vector(0)

Question:
1. Is this fix is acceptable? If yes, I will update the other related dashboard
2. Any script to run after updating the templates? I ran 'generate-dashboard.sh', it updated the files in grafana/build

c0ffeec0der commented:
Hi guys, any comment? I'd be happy to update the other templates :)

amnonh commented:
@c0ffeec0der thanks for the PR.
please add an issue this PR is closing, also, please unless github made me miss it, I didn't see the generated dashboard.

c0ffeec0der commented:
Issue created, #976 

The generated dashboard:

- grafana/build/ver_3.3/scylla-overview.3.3.json 
- grafana/build/ver_4.0/scylla-overview.4.0.json

Question: 
- other than running 'generate-dashboard.sh', any other script to run to generate dashboard?
- Is this fix acceptable? I will then proceed to fix the same panel on other dashboard

amnonh commented:
You should run the ./generate-dashboard.sh per version:

`./generate-dashboard.sh  -v master,4.0,4.1,3.3,2019.1,2018.1,3.2`

Commit:
Joining stat

Problem :
does not show new pod joining

How to simulate:
1. Deploy using scylla operator for 1 pod
2. Add 1 pod

Root cause:
Arithmetic addition in prometheus expr is not working if one condition is empty. Only work when both condition return value.

Fix:

Original code
>count(scylla_node_operation_mode==1) +  count(scylla_node_operation_mode==2)OR vector(0)

Fixing code
>count(scylla_node_operation_mode<=2)OR vector(0)

c0ffeec0der commented:
Ok, done. The other dashboard that have panel 'joining' is 'alternator'. Ran generate-dashboards.sh for all version
